### ISSUE TITLE
[Bug]: Extensions disappear after vscode update

### ISSUE THREAD

**Issue by rogerlga** [ScoopInstaller/Extras#10013]
### Prerequisites

- [X] I have written a descriptive issue title.
- [X] I have searched all issues/PRs to ensure it has not already been reported or fixed.
- [X] I have verified that I am using the latest version of Scoop and corresponding bucket.

### Package Name

vscode

### Expected/Current Behaviour

Extensions disappeared after update. It seems that <sccop_dir>\persist\vscode\data\extensions\extensions.json has paths containing the old version of vscode (1.74.0 instead of 1.74.1 in my case):

```txt
[{"identifier":{"id":"ms-toolsai.jupyter-keymap","uuid":"9f6dc8db-620c-4844-b8c5-e74914f1be27"},"version":"1.0.0","location":{"$mid":1,"fsPath":"c:\\Users\\hr73\\scoop\\apps\\vscode\\1.74.0\\data\\extensions\\ms-toolsai.jupyter-keymap-1.0.0","_sep":1,"path":"/c:/Users/hr73/scoop/apps/vscode/1.74.0/data/extensions/ms-toolsai.jupyter-keymap-1.0.0","scheme":"file"},"metadata":{"id":"9f6dc8db-620c-4844-b8c5-e74914f1be27","publisherId":"ac8eb7c9-3e59-4b39-8040-f0484d8170ce","publisherDisplayName":"Microsoft","targetPlatform":"undefined","isApplicationScoped":false,"updated":false,"isPreReleaseVersion":false,"installedTimestamp":1671067056174,"preRelease":false}},{"identifier":{"id":"ms-toolsai.vscode-jupyter-slideshow","uuid":"e153ca70-b543-4865-b4c5-b31d34185948"},"version":"0.1.5","location":{"$mid":1,"fsPath":"c:\\Users\\hr73\\scoop\\apps\\vscode\\1.74.0\\data\\extensions\\ms-toolsai.vscode-jupyter-slideshow-0.1.5","_sep":1,"path":"/c:/Users/hr73/scoop/apps/vscode/1.74.0/data/extensions/ms-toolsai.vscode-jupyter-slideshow-0.1.5","scheme":"file"},"metadata":{"id":"e153ca70-b543-4865-b4c5-b31d34185948","publisherId":"ac8eb7c9-3e59-4b39-8040-f0484d8170ce","publisherDisplayName":"Microsoft","targetPlatform":"undefined","isApplicationScoped":false,"updated":false,"isPreReleaseVersion":false,"installedTimestamp":1671067056184,"preRelease":false}},{"identifier":{"id":"ms-toolsai.vscode-jupyter-cell-tags","uuid":"ab4fb32a-befb-4102-adf9-1652d0cd6a5e"},"version":"0.1.6","location":{"$mid":1,"fsPath":"c:\\Users\\hr73\\scoop\\apps\\vscode\\1.74.0\\data\\extensions\\ms-toolsai.vscode-jupyter-cell-tags-0.1.6","_sep":1,"path":"/c:/Users/hr73/scoop/apps/vscode/1.74.0/data/extensions/ms-toolsai.vscode-jupyter-cell-tags-0.1.6","scheme":"file"},"metadata":{"id":"ab4fb32a-befb-4102-adf9-1652d0cd6a5e","publisherId":"ac8eb7c9-3e59-4b39-8040-f0484d8170ce","publisherDisplayName":"Microsoft","targetPlatform":"undefined","isApplicationScoped":false,"updated":false,"isPreReleaseVersion":false,"installedTimestamp":1671067056191,"preRelease":false}},{"identifier":{"id":"ms-toolsai.jupyter-renderers","uuid":"b15c72f8-d5fe-421a-a4f7-27ed9f6addbf"},"version":"1.0.12","location":{"$mid":1,"fsPath":"c:\\Users\\hr73\\scoop\\apps\\vscode\\1.74.0\\data\\extensions\\ms-toolsai.jupyter-renderers-1.0.12","_sep":1,"path":"/c:/Users/hr73/scoop/apps/vscode/1.74.0/data/extensions/ms-toolsai.jupyter-renderers-1.0.12","scheme":"file"},"metadata":{"id":"b15c72f8-d5fe-421a-a4f7-27ed9f6addbf","publisherId":"ac8eb7c9-3e59-4b39-8040-f0484d8170ce","publisherDisplayName":"Microsoft","targetPlatform":"undefined","isApplicationScoped":false,"updated":false,"isPreReleaseVersion":false,"installedTimestamp":1671067056179,"preRelease":false}},{"identifier":{"id":"ms-toolsai.jupyter","uuid":"6c2f1801-1e7f-45b2-9b5c-7782f1e076e8"},"version":"2022.11.1003412109","location":{"$mid":1,"fsPath":"c:\\Users\\hr73\\scoop\\apps\\vscode\\1.74.0\\data\\extensions\\ms-toolsai.jupyter-2022.11.1003412109","_sep":1,"path":"/c:/Users/hr73/scoop/apps/vscode/1.74.0/data/extensions/ms-toolsai.jupyter-2022.11.1003412109","scheme":"file"},"metadata":{"id":"6c2f1801-1e7f-45b2-9b5c-7782f1e076e8","publisherId":"ac8eb7c9-3e59-4b39-8040-f0484d8170ce","publisherDisplayName":"Microsoft","targetPlatform":"undefined","isApplicationScoped":false,"updated":false,"isPreReleaseVersion":false,"installedTimestamp":1671067056169,"preRelease":false}},{"identifier":{"id":"ms-python.python","uuid":"f1f59ae4-9318-4f3c-a9b5-81b2eaa5f8a5"},"version":"2022.20.1","location":{"$mid":1,"fsPath":"c:\\Users\\hr73\\scoop\\apps\\vscode\\1.74.0\\data\\extensions\\ms-python.python-2022.20.1","_sep":1,"path":"/c:/Users/hr73/scoop/apps/vscode/1.74.0/data/extensions/ms-python.python-2022.20.1","scheme":"file"},"metadata":{"id":"f1f59ae4-9318-4f3c-a9b5-81b2eaa5f8a5","publisherId":"998b010b-e2af-44a5-a6cd-0b5fd3b9b6f8","publisherDisplayName":"Microsoft","targetPlatform":"undefined","isApplicationScoped":false,"updated":false,"isPreReleaseVersion":false,"installedTimestamp":1671067056160,"preRelease":false}},{"identifier":{"id":"ms-python.isort","uuid":"4ad0ce32-ff3f-49f0-83b5-93e5dc00cfff"},"version":"2022.8.0","location":{"$mid":1,"fsPath":"c:\\Users\\hr73\\scoop\\apps\\vscode\\1.74.0\\data\\extensions\\ms-python.isort-2022.8.0","_sep":1,"path":"/c:/Users/hr73/scoop/apps/vscode/1.74.0/data/extensions/ms-python.isort-2022.8.0","scheme":"file"},"metadata":{"id":"4ad0ce32-ff3f-49f0-83b5-93e5dc00cfff","publisherId":"998b010b-e2af-44a5-a6cd-0b5fd3b9b6f8","publisherDisplayName":"Microsoft","targetPlatform":"undefined","isApplicationScoped":false,"updated":false,"isPreReleaseVersion":false,"installedTimestamp":1671067066524,"preRelease":false}},{"identifier":{"id":"ms-python.vscode-pylance","uuid":"364d2426-116a-433a-a5d8-a5098dc3afbd"},"version":"2022.12.20","location":{"$mid":1,"fsPath":"c:\\Users\\hr73\\scoop\\apps\\vscode\\1.74.0\\data\\extensions\\ms-python.vscode-pylance-2022.12.20","_sep":1,"path":"/c:/Users/hr73/scoop/apps/vscode/1.74.0/data/extensions/ms-python.vscode-pylance-2022.12.20","scheme":"file"},"metadata":{"id":"364d2426-116a-433a-a5d8-a5098dc3afbd","publisherId":"998b010b-e2af-44a5-a6cd-0b5fd3b9b6f8","publisherDisplayName":"Microsoft","targetPlatform":"undefined","isApplicationScoped":false,"updated":false,"isPreReleaseVersion":false,"installedTimestamp":1671067066511,"preRelease":false}},{"identifier":{"id":"twxs.cmake","uuid":"2be7cf14-2603-402e-9771-fd79df83cdf8"},"version":"0.0.17","location":{"$mid":1,"fsPath":"c:\\Users\\hr73\\scoop\\apps\\vscode\\1.74.0\\data\\extensions\\twxs.cmake-0.0.17","_sep":1,"path":"/c:/Users/hr73/scoop/apps/vscode/1.74.0/data/extensions/twxs.cmake-0.0.17","scheme":"file"},"metadata":{"id":"2be7cf14-2603-402e-9771-fd79df83cdf8","publisherId":"b5ab7991-81a8-4557-af54-4cb8e01e85c1","publisherDisplayName":"twxs","targetPlatform":"undefined","isApplicationScoped":false,"updated":false,"isPreReleaseVersion":false,"installedTimestamp":1671067286860,"preRelease":false}},{"identifier":{"id":"ms-vscode.cmake-tools","uuid":"7c889349-8749-43d4-8b5e-08939936d7f4"},"version":"1.12.27","location":{"$mid":1,"fsPath":"c:\\Users\\hr73\\scoop\\apps\\vscode\\1.74.0\\data\\extensions\\ms-vscode.cmake-tools-1.12.27","_sep":1,"path":"/c:/Users/hr73/scoop/apps/vscode/1.74.0/data/extensions/ms-vscode.cmake-tools-1.12.27","scheme":"file"},"metadata":{"id":"7c889349-8749-43d4-8b5e-08939936d7f4","publisherId":"5f5636e7-69ed-4afe-b5d6-8d231fb3d3ee","publisherDisplayName":"Microsoft","targetPlatform":"undefined","isApplicationScoped":false,"updated":false,"isPreReleaseVersion":false,"installedTimestamp":1671067287353,"preRelease":false}},{"identifier":{"id":"tht13.rst-vscode","uuid":"9a76007e-6367-44f0-b175-2439b4e81196"},"version":"3.0.1","location":{"$mid":1,"fsPath":"c:\\Users\\hr73\\scoop\\apps\\vscode\\1.74.0\\data\\extensions\\tht13.rst-vscode-3.0.1","_sep":1,"path":"/c:/Users/hr73/scoop/apps/vscode/1.74.0/data/extensions/tht13.rst-vscode-3.0.1","scheme":"file"},"metadata":{"id":"9a76007e-6367-44f0-b175-2439b4e81196","publisherId":"78ff7982-87b9-4f5c-8886-f79f581462f0","publisherDisplayName":"Thomas Haakon Townsend","targetPlatform":"undefined","isApplicationScoped":false,"updated":false,"isPreReleaseVersion":false,"installedTimestamp":1671067435219,"preRelease":false}},{"identifier":{"id":"trond-snekvik.simple-rst","uuid":"ff749cb0-bd02-4ca4-9f69-d6157655ba32"},"version":"1.5.2","location":{"$mid":1,"fsPath":"c:\\Users\\hr73\\scoop\\apps\\vscode\\1.74.0\\data\\extensions\\trond-snekvik.simple-rst-1.5.2","_sep":1,"path":"/c:/Users/hr73/scoop/apps/vscode/1.74.0/data/extensions/trond-snekvik.simple-rst-1.5.2","scheme":"file"},"metadata":{"id":"ff749cb0-bd02-4ca4-9f69-d6157655ba32","publisherId":"c2e3f2a2-1986-4784-a2bd-308925f721d7","publisherDisplayName":"Trond Snekvik","targetPlatform":"undefined","isApplicationScoped":false,"updated":false,"isPreReleaseVersion":false,"installedTimestamp":1671067481732,"preRelease":false}},{"identifier":{"id":"ms-vscode.cpptools","uuid":"690b692e-e8a9-493f-b802-8089d50ac1b2"},"version":"1.13.8","location":{"$mid":1,"fsPath":"c:\\Users\\hr73\\scoop\\apps\\vscode\\1.74.0\\data\\extensions\\ms-vscode.cpptools-1.13.8-win32-x64","_sep":1,"path":"/c:/Users/hr73/scoop/apps/vscode/1.74.0/data/extensions/ms-vscode.cpptools-1.13.8-win32-x64","scheme":"file"},"metadata":{"id":"690b692e-e8a9-493f-b802-8089d50ac1b2","publisherId":"5f5636e7-69ed-4afe-b5d6-8d231fb3d3ee","publisherDisplayName":"Microsoft","targetPlatform":"win32-x64","isApplicationScoped":false,"updated":true,"isPreReleaseVersion":false,"installedTimestamp":1671181519488,"preRelease":false}}]
```

All the extensions still remain in the folder, but vscode can not load then:

![image](https://user-images.githubusercontent.com/9919528/208066166-17cc27a9-beda-4d0b-aa4f-b38ba9764535.png)

![image](https://user-images.githubusercontent.com/9919528/208066350-b6e8614a-69e0-499c-b1ca-ab8778e1b44b.png)

### Steps to Reproduce

```console
PS> scoop update vscode
PS> scoop cleanup vscode --cache

After update (1.74.0 to 1.74.1 in my case) and cleanup old version, then open vscode. Previous installed extensions aren't loaded anymore.
```

### Possible Solution

Do some work with extensions.json?

### Scoop and Buckets Version

```console
PS> scoop --version
Current Scoop version:
v0.3.1 - Released at 2022-11-15

'extras' bucket:
16986b8bf (HEAD -> master, origin/master, origin/HEAD) anki: Update to version 2.1.55

'main' bucket:
b4ed35b81 (HEAD -> master, origin/master, origin/HEAD) x265: Update to version 3.5+73-a7ca459

'versions' bucket:
6aef044de (HEAD -> master, origin/master, origin/HEAD) zig-dev: Update to version 0.11.0-dev.815
```

### Scoop Config

```console
PS> scoop config

last_update                       scoop_branch scoop_repo
-----------                       ------------ ----------
2022-12-16T06:09:46.4196558-03:00 master       https://github.com/ScoopInstaller/Scoop
```

### PowerShell Version

```console
PS> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.19041.1682
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.19041.1682
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

**Comment by slqy123** [comment]
in `data/extensions/extensions.json`
```json
        "location": {
            "$mid": 1,
            "fsPath": "c:\\Users\\20577\\Scoop\\apps\\vscode\\1.74.0\\data\\extensions\\vue.volar-1.0.13-win32-x64",
            "_sep": 1,
            "path": "/c:/Users/20577/Scoop/apps/vscode/1.74.0/data/extensions/vue.volar-1.0.13-win32-x64",
            "scheme": "file"
        },
```
As you can see, it uses `1.74.0/data`, which you may have cleaned up. So you can just replace `1.74.0` with `current` or create a symlink of `1.74.0`

**Comment by rogerlga** [comment]
This is the workaround that I was using, but we'll have this issue everytime we update and install new extension or update extensions, as it writes the current vscode version in the paths inside `extensions.json`, instead of "current".

**Comment by Zliced13** [comment]
It's also a VSCodium issue as well. See https://github.com/microsoft/vscode/issues/169359 for more.

**Comment by Zliced13** [comment]
![](https://user-images.githubusercontent.com/9919528/208066166-17cc27a9-beda-4d0b-aa4f-b38ba9764535.png)
@rogerlga I did this as a workaround, DO NOT DOWNGRADE. I would start by closing VSCode, and removing all the extensions in your `persist/vscode/data/extensions` folder. After that, open VSCode and start manually re-installing all your extensions.

**Comment by g-berthiaume** [comment]
Same problem here.

**Comment by g-berthiaume** [comment]
While I would not describe @Zliced13 workaround as a solution, it did work for me.
Let us know if you find a way to not delete all our extensions.

I can't imagine how painful it must be for somebody living in a country without fast internet.

**Comment by Zliced13** [comment]
Yeah, I agree with @g-berthiaume here. If anybody has a better alternative to my workaround, I would love to see it.

**Comment by otavio-silva** [comment]
Was just about to create an issue here.
This is the third time all my extensions vanished and I have to install everything again....

**Comment by antoinemartin** [comment]
You don't need to remove all extensions. The following powershell snippet fixes the path of the extensions in the `extensions.json` file:

```console
PS>  (Get-Content $env:USERPROFILE\scoop\persist\vscode\data\extensions\extensions.json).replace('1.74.1','1.74.2') | Set-Content $env:USERPROFILE\scoop\persist\vscode\data\extensions\extensions.json
```

**Comment by antoinemartin** [comment]
I made a [PR to fix the issue](https://github.com/ScoopInstaller/Extras/pull/10074). I have tested it with the `1.74.1` to `1.74.2` update and it worked for me.

**Comment by antoinemartin** [comment]
I badly copy/pasted my fix for vscode in #10074 .  I have made a new PR: #10089. Sorry about that.
